src/udp_socket.cpp

```diff
@@ -311,6 +311,16 @@ void udp_socket::on_read(error_code const& ec, udp::socket* s)
 
 		if (err == boost::asio::error::would_block || err == boost::asio::error::try_again) break;
 		on_read_impl(ep, err, bytes_transferred);
+
+		// found on iOS, socket will be disconnected when app goes backgroud. try to reopen it.
+		if (err == boost::asio::error::not_connected) 
```

Comment:
could you check for boost::system::errc::bad_file_descriptor too? My understanding is that it's not completely defined which error you'll get in this scanario, but it's likely to be one of these two.